azurelinuxagent/ga/exthandlers.py

```diff
@@ -416,7 +425,11 @@ def handle_ext_handler(self, ext_handler, etag):
 
             state = ext_handler.properties.state
 
-            if ext_handler_i.decide_version(target_state=state) is None:
+            # The Guest Agent currently only supports 1 installed version per extension on the VM.
+            # If the extension version is unregistered and the customers wants to uninstall the extension,
+            # we should let it go through even if the installed version doesnt exist in Handler manifest (PIR) anymore.
+            # If target state is enabled and version not found in manifest, do not process the extension.
```

Comment:
> If target state is enabled and version not found in manifest, do not process the extension.
 
this should be flagged as an error rather than ignored, right?

Reply:
An error is logged inside the `def decide_version` function. Here's the exact log line - 

```python
f target_state in (ExtensionRequestedState.Uninstall, ExtensionRequestedState.Disabled):
            if installed_pkg is None:
                msg = "Failed to find installed version: {0} of Handler: {1}  in handler manifest to uninstall.".format(
                    installed_version, self.ext_handler.name)
                self.logger.warn(msg)
```